clang/lib/AST/DeclPrinter.cpp

```diff
@@ -855,7 +868,17 @@ void DeclPrinter::VisitFunctionDecl(FunctionDecl *D) {
         Out << Proto << " -> ";
         Proto.clear();
       }
-      AFT->getReturnType().print(Out, Policy, Proto);
+      if (!Policy.SuppressTagKeyword && Policy.SuppressScope &&
+          !Policy.SuppressUnwrittenScope) {
+        AddPrefix(Policy, AFT->getReturnType(), Out);
+        bool OldTagKeyword = Policy.SuppressTagKeyword;
+        bool OldSupressScope = Policy.SuppressScope;
+        AFT->getReturnType().print(Out, Policy, Proto);
+        Policy.SuppressTagKeyword = OldTagKeyword;
+        Policy.SuppressScope = OldSupressScope;
```

Comment:
This is only necessary because of the changes you made in TypePrinter.cpp, right? e.g., if we reset the policy back to its original values there, we don't need to do it here.